app/models/cda/prosecution_case_search.rb

```diff
@@ -0,0 +1,10 @@
+module Cda
+  class ProsecutionCaseSearch < BaseModel
+    self.element_name = "prosecution_case"
+
+    def self.execute(filter_params)
+      http_response = post('/', nil, { filter: filter_params }.to_json)
+      ProsecutionCaseSearchResponse.new(JSON.parse(http_response.body))
+    end
+  end
+end
```

Comment:
I am bit confused about this class,
it is a model, but it behaves like a service, 
it uses post but it is actually not creating a new resource 🤔 
... I understand this comment is not helpful, just I am not totally sure about the needs of this refactor,
there wasn't another way to avoid put PII into logs?